### PR TITLE
Ft blogs

### PR DESCRIPTION
**Pull request by FelixWambiri**
*What does this PR do?*

User can be able to add blogs to the healthchecks website

*Description of Task to be completed?*
• User can be able to add a new blog posts categories
• User can be able to add a new blog post to the site
• User can be able to edit their blog post previously added
• User can be able to delete their blogposts
• User can be able to publicly share blog posts

*How should this be manually tested?*

• Clone the repository `https://github.com/andela/hc-bigas-opus`
• Change Directory to the project
• Create a virtual environment
• Activate the virtual environment
• Run migrations `python manage.py migrate`  
   Navigate to `http://127.0.0.1:8000/` to login then proceed to admin `http://127.0.0.1:8000/admin/`

*What are the relevant pivotal tracker stories?*

•155965586

*Screenshots (if appropriate)*
screen shot 2018-04-12 at 22 05 54
screen shot 2018-04-12 at 22 06 35
screen shot 2018-04-12 at 22 06 06
screen shot 2018-04-12 at 22 06 22

### CODE REVIEW

File: hc/blogs/views.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    }
+    if request.method == 'POST':
+        if "create_category" in request.POST:
+            if form.is_valid():
+                category = form.save(commit=False)
+                category.title = form.cleaned_data['title']
+                try:
+                    category.save()
+                except IntegrityError:
+                    messages.warning(request, 'The category already exists.Create a blog post under that category!')
```
Comment (Kalela) [inline hc/blogs/views.py:41]: Good implementation of dynamic categories.

File: static/css/blog.css [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    margin-left: 8%;
+    font-size: 47px;
+    font-family: 'Open Sans Condensed', sans-serif;
+}
+#post{
+    font-family: 'Slabo 27px', serif;
+}
+#post1{
+    font-family: 'Slabo 27px', serif;
+}
```
Comment (Kalela) [inline static/css/blog.css:133]: The pages styling looks good. Follows the healthchecks theme.

**Comment by betsybeth** [comment]
Good work.

File: hc/blogs/views.py [hunk 3]
```diff
@@ -34,8 +35,10 @@ def create_categories_and_blogs(request):
             if form.is_valid():
                 category = form.save(commit=False)
                 category.title = form.cleaned_data['title']
-                category.save()
-                return render(request, 'blog/create_categories_and_blogs.html', cxt)
+                try:
```
Comment (betsybeth) [inline hc/blogs/views.py]: Good implementation on catching errors.

**Comment by mutoromike** [comment]
Great implementation Felix! Good work with the UI too. Maybe next time if you could just try and increase the test coverage it will be better. You have also done a good job in commenting your methods and classes. An example is given below:

screen shot 2018-04-13 at 11 12 19

**Review by frankip — COMMENTED** [review]
good implementation of the slug generation

File: hc/blogs/tests.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        self.assertEqual('read', category.title)
+
+
+    def test_home_page_returns_all_categories(self):
+        url = reverse('blogs:hc-category')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'blog/view_blogs.html')
+
+    # def test_view_blogs__by_category(self):
```
Comment (frankip) [inline hc/blogs/tests.py:46]: remove the commented out files

**Review by meshnesh — APPROVED** [review]
Good work creating the blog app and running the tests

File: hc/blogs/views.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            category_input = request.POST['selectop']
+            content = request.POST['content']
+            author = request.user
+            if title and category_input and content:
+                category_query = get_object_or_404(BlogPostsCategory, title=category_input)
+                blog=BlogPosts(author=author, title=title, content=content, category=category_query)
+                blog.save()  
+                return HttpResponseRedirect('/blogs/', cxt)
+        return render(request, 'blog/create_categories_and_blogs.html', cxt)
+        # return HttpResponseRedirect('/blogs/', cxt)
```
Comment (frankip) [inline hc/blogs/views.py:55]: remove comments

File: hc/blogs/views.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        return render(request, "blog/view_blogs.html", cxt)
+    message = "No Blog Stories In This Category"
+    cxt = {
+        'categories':categories,
+        'message': message
+        }
+    return render(request, 'blog/view_blogs.html', cxt)
+
+@login_required
+def add_comment(request, post):
```
Comment (frankip) [inline hc/blogs/views.py:94]: good job

**Review by frankip — CHANGES_REQUESTED** [review]
overall great implementation and good work

**Pull request merged**